### DISCUSSION TITLE
model.predict

### DISCUSSION THREAD

**Discussion by KavyashreeRai** [ultralytics/ultralytics#8432] [Q&A]
Am using yolov8 when i try to run the inference through model.predict it takes a long time to load. am passing a live image stream in a particular window by the time the inference/prediction starts the subject would have moved outside the window of capture. can you pls let me know what are the possible solutions to preload the model.

**Comment by pderrenger** [comment]
Hello! It seems like you're experiencing latency during the inference stage with YOLOv8, especially when processing a live image stream. To reduce the time it takes for `model.predict` to start making predictions, you can ensure the model is preloaded and ready to infer as soon as the stream begins. Here's a quick example of how you might preload your model:

```python
from ultralytics import YOLO

# Preload the model
model = YOLO('yolov8n.pt', preload=True)

# Now the model is ready to predict
# You can start your image stream and call model.predict() on each frame
```

By setting `preload=True`, the model and its weights are loaded into memory before the stream starts, which should help reduce the initial latency. If you're still facing delays, consider checking your hardware utilization to ensure resources are not being bottlenecked. Additionally, using a smaller model or reducing the input image size can also speed up inference times.

Hope this helps! Let us know if you have any more questions. 😊🚀

**Reply by KavyashreeRai** [reply]
We are using Nvidia Jetson Nano for inference.. The objects are captured in 3 dimensions. There are 3 systems and in each dimension,,, there is a system.

**Reply by KavyashreeRai** [reply]
Thx Paula for your kind reply.. I missed checking the arguments or it slipped.. Thx again

**Reply by pderrenger** [reply]
@KavyashreeRai you're welcome! If you have any more questions or need further assistance, just let us know. Happy coding! 😊👍
